src/client/app/components/meters/EditMeterModalComponent.tsx

```diff
@@ -240,6 +243,22 @@ export default function EditMeterModalComponent(props: EditMeterModalComponentPr
 				props.meter.maxError != state.maxError ||
 				props.meter.disableChecks != state.disableChecks
 			);
+		let error_message = '';
+		for (const value of Object.values(groups)) {
+			for (let i = 0; i < value.deepMeters.length; i++) {
+				console.log(value.deepMeters[i], props.meter.id)
+				if(value.deepMeters[i]== props.meter.id)
+				{
+					error_message += translate('group') + value.name + ' uses ' + translate('meter') + ' "' + metersByID[value.deepMeters[i]].name + '" \n';
+				}
+			}
+		}
+		state.unitId = props.meter.unitId;
```

Comment:
First, I'm not sure why the state is always being reset. Second, directly setting state can cause issues. I think using setState() is the way to go here.